app/views/hosts/_form.html.erb

```diff
@@ -134,6 +134,6 @@
 
     <%= hidden_field_tag 'bare_metal_capabilities', @host.bare_metal_capabilities %>
     <%= f.hidden_field :overwrite? %>
-    <%= submit_or_cancel f, @host.overwrite?, :cancel_path => @host.new_record? || !@host.name_changed? ? @host : edit_host_path(@host.name_was) %>
+    <%= submit_or_cancel f, @host.overwrite?, :cancel_path => (@host.new_record? ? hosts_path : host_path(:id => @host.id)) %>
```

Comment:
When clicking the cancel button on an existing host, this gives me `http://localhost:3000/hosts/26` instead of `http://localhost:3000/hosts/raul-gherardi.example.com`. Unfortunately the name change handling in the original code is really necessary.
Simply using `host_path(:id => @host.name_was)` should work in any case. I can't see any sense in the `edit_host_path` in the original code.